Content.Shared/Emag/Systems/EmagSystem.cs

```diff
@@ -50,6 +52,15 @@ public bool TryUseEmag(EntityUid uid, EntityUid user, EntityUid target, EmagComp
         if (_tag.HasTag(target, comp.EmagImmuneTag))
             return false;
 
+        // DeltaV - Add a whitelist / blacklist to the Emag
+        if (_whitelistSystem.IsWhitelistFail(comp.Whitelist, target)
+            || _whitelistSystem.IsWhitelistPass(comp.Blacklist, target))
+        {
+            _popup.PopupClient("Invalid target!", user, user); // do i really need to use a locale for this?
```

Comment:
I suggest adding a locale string `emag-invalid-target = {$emag} has no effect on ${target}` and replacing this with `_popup.PopupClient(Loc.GetString("emag-invalid-target", ("emag", uid), ("target", target)), user, user);`.

Reply:
done :)